scripts/xtensa-build-zephyr.py

```diff
@@ -1169,6 +1175,36 @@ def reproducible_checksum(platform, ri_file):
 	chk256 = sof_ri_info.EraseVariables(ri_file, parsed_ri, west_top / repro_output)
 	print('sha256sum {0}\n{1} {0}'.format(repro_output, chk256))
 
+def create_tarball():
+	# vendor directories (typically under /lib/firmware) to include
+        # in the tarball
+	vendor_dirs = [ 'intel']
+
+	build_top = west_top / 'build-sof-staging' / 'sof'
+
+	# Some CI/build usages incrementally call this script to create
+	# a staging area with multiple targets. Support this mode
+	# of use by naming the tarball with target name in case only a single
+	# target is built. In other cases, use a generic tarball name.
```

Comment:
I still don't get this sorry. If you build targets "incrementally" without cleaning then recreating the tarball from scratch every time will "incrementally" add previously built targets: no need for this special case.

Also, why would building targets 2 by 2 or 3 by 3 differ from 1 by 1?

This really look like unnecessary, extra complexity to me.

Reply:
@marc-hb The build scripts we use to prepare Intel binaries for release, build a full set of platforms calling this script one platform at a time. After each build, files are copied out to storage and the work area is cleaned before next platform is built. Thus if a single "sof.tgz" filename is used in this script as output, at the end of this release build creation, there's a sof.tgz with only one platform (the last one built). We cannot incrementally update a tarball that was already uploaded to strorage (with the same name).

With current version of this script, "sof-FOO.tgz" is created when bulding for FOO, and uploaded at each iteration. 

For deployment, this is not ideal, so I'd like to keep the option to run this script to build a full set of platforms, and create a singular tarball that contains all files for the release. So if/when we change the Intel release scripts, and/or somebody else takes this script to prepare non-Intel binaries, they can build full set of platforms in one go and get a nice single tarball as a result.

But yeah, it's a bit ugly but not sure how to change. Enumerating all platforms and building a tarball for each one in this function is not easy as if this script is called with e.g. "tgl tgl-h", we don't really know which files need to be in which tarball (without hardcoding Intel-specific paths to search for).